### PR TITLE
boards: arm: add the Seeeduino XIAO board

### PR DESCRIPTION
**Pull request by nzmichaelh**
The Seeeduino XIAO is a tiny (20 mm x 17.5 mm) ARM development
board with onboard LEDs, USB port, and range of I/O broken out
onto 14 pins.

### CODE REVIEW

File: boards/arm/seeeduino_xiao/board.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2020 Google LLC.
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(bossac "--offset=0x2000")
```
Comment (stephanosio) [inline boards/arm/seeeduino_xiao/board.cmake:4]: Reminder that this needs to be removed if https://github.com/zephyrproject-rtos/zephyr/pull/26510 is merged before this PR (and vice versa).

File: boards/arm/seeeduino_xiao/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+#. Flash the image:
+
+   .. zephyr-app-commands::
+      :zephyr-app: samples/hello_world
+      :board: seeeduino_xiao
+      :goals: flash
+      :compact:
+
+   You should see "Hello World! arm" in your terminal.
```
Comment (stephanosio) [inline boards/arm/seeeduino_xiao/doc/index.rst:144]: This is no longer the case.

The `hello_world` sample now prints the board name:
https://github.com/zephyrproject-rtos/zephyr/blob/f4896adaa86cef3bfb4c2580e1d880f0718da01a/samples/hello_world/src/main.c#L12

File: boards/arm/seeeduino_xiao/pinmux.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		device_get_binding(DT_LABEL(DT_NODELABEL(pinmux_a)));
+	struct device *muxb =
+		device_get_binding(DT_LABEL(DT_NODELABEL(pinmux_b)));
+
+	ARG_UNUSED(dev);
+	/* Prevent an unused variable error if all peripherals are disabled */
+	(void)muxa;
+	(void)muxb;
+
+#if (ATMEL_SAM0_DT_SERCOM_CHECK(4, atmel_sam0_uart) && CONFIG_UART_SAM0)
```
Comment (stephanosio) [inline boards/arm/seeeduino_xiao/pinmux.c:23]:
```suggestion
#if (ATMEL_SAM0_DT_SERCOM_CHECK(4, atmel_sam0_uart) && defined(CONFIG_UART_SAM0))
```

or alternatively:
```suggestion
#if (ATMEL_SAM0_DT_SERCOM_CHECK(4, atmel_sam0_uart) && IS_ENABLED(CONFIG_UART_SAM0))
```

File: boards/arm/seeeduino_xiao/board.cmake (hunk 1 above)
Comment (nzmichaelh) [reply boards/arm/seeeduino_xiao/board.cmake:4]: Ack.  I have a few SAMD related PRs in flight at the moment and I'll update them as they land.

File: boards/arm/seeeduino_xiao/doc/index.rst (hunk 2 above)
Comment (nzmichaelh) [reply boards/arm/seeeduino_xiao/doc/index.rst:144]: Done

File: boards/arm/seeeduino_xiao/pinmux.c (hunk 3 above)
Comment (nzmichaelh) [reply boards/arm/seeeduino_xiao/pinmux.c:23]: Done

**Review by stephanosio — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**